### PR TITLE
common: use new rust-based switch script

### PR DESCRIPTION
**Pull request by Mic92**
This implementation is meant to replace switch-to-configuration.pl with a rust version.
I tested it on my infra for now.

### CODE REVIEW

**Comment by zimbatm** [comment]
Looks like it's not available on aarch64 yet?

**Comment by Mic92** [comment]
Fixed, but we need to disable it on stable.

File: nixos/common/default.nix [hunk 1]
```diff
@@ -16,6 +16,13 @@
     ./zfs.nix
   ];
 
+  system.switch = {
+    enable = lib.mkDefault false;
```
Comment (zimbatm) [inline nixos/common/default.nix:20]: @Mic92 what happens when this is set to false and enableNg is unset?

Comment (SuperSandro2000) [reply nixos/common/default.nix:20]: Then you can't switch. That's what the perlless tests are using.

Comment (Mic92) [reply nixos/common/default.nix:20]: Ah hang on. This is actually pretty bad if you are on an old version.

Comment (Mic92) [reply nixos/common/default.nix:20]: Fixed in https://github.com/nix-community/srvos/pull/433/files

**Pull request merged**